hi-ml-azure/src/health_azure/utils.py

```diff
@@ -480,25 +480,29 @@ def __init__(self, run_id: str, checkpoint_filename: str, azure_config_json_path
         :param aml_workspace: An optional Azure ML Workspace object. If not running inside an AML Run, and no
             azure_config_json_path is provided, this is required.
         :param download_dir: The local directory in which to save the downloaded checkpoint files.
-        :param remote_checkpoint_folder: The remote folder from which to download the checkpoint file
+        :param remote_checkpoint_dir: The remote folder from which to download the checkpoint file
         """
         self.azure_config_json_path = azure_config_json_path
         self.aml_workspace = aml_workspace
         self.run_id = run_id
         self.checkpoint_filename = checkpoint_filename
         self.download_dir = Path(download_dir)
-        self.remote_checkpoint_folder = Path(remote_checkpoint_folder)
+        self.remote_checkpoint_dir = Path(remote_checkpoint_dir)
 
     @property
-    def local_checkpoint_path(self) -> Path:
+    def local_checkpoint_dir(self) -> Path:
         # in case we run_id is a run recovery id, extract the run id
         run_id_parts = self.run_id.split(":")
         run_id = run_id_parts[-1]
-        return self.download_dir / run_id / self.checkpoint_filename
+        return self.download_dir / run_id 
 
     @property
     def remote_checkpoint_path(self) -> Path:
-        return self.remote_checkpoint_folder / self.checkpoint_filename
+        return self.remote_checkpoint_dir / self.checkpoint_filename
+    
+    @property
+    def local_checkpoint_path(self) -> Path:
+        return self.local_checkpoint_dir / self.remote_checkpoint_path
```

Comment:
this is the core of the change - the underlying function is downloading the full hierarchy of folders from the run